Inlet docs for simple types (#408)
LLNL/axom: src/axom/inlet/docs/sphinx/index.rst

------------

> Inlet provides an easy and extensible way to handle input files for simulation code.

Reviewer: ```suggestion
Inlet provides an easy and extensible way to handle input files for a simulation code.
```

> We provide JSON, Lua, and YAML functionality but additional

Reviewer: Minor:
```suggestion
We provide readers for JSON, Lua, and YAML. Additional
```

languages can be supported via an implementation of Inlet's Reader interface.

All information read from the input file is stored via Inlet into a user-provided Sidre

> DataStore.  This allows us to utilize the functionality of Sidre, such as simulation

Reviewer: 👍

------------

Inlet provides an easy and extensible way to handle input files for simulation code.

> We provide JSON, Lua, and YAML functionality but any language can be used via an inherited

Reviewer: Suggestion: `any language` -> `other languages` or `additional languages` ?

Author: additional languages

> Reader class. Inlet is used to define the structure of the information expected in your input file.

Reviewer 2: Minor: Maybe something like "via an implementation of the Reader interface" would be a little clearer?
